test/functional/cmdline_options_tester/src/Test.java

```diff
@@ -546,7 +549,10 @@ public ProcessKiller(Process proc, long timeout, String exeToDebug) {
 		public synchronized void run() {
 			final long endTime = System.currentTimeMillis() + _procTimeout;
 			while (_proc.isAlive()) {
-				long currTimeout = endTime - System.currentTimeMillis();
+				/* Set up a fixed timer without timeout for test on RISC-V
+				 * as it is pretty slow without JIT support for now.
+				 */
+				long currTimeout = (isRiscv) ? _procTimeout : (endTime - System.currentTimeMillis());
```

Comment:
Is it better to change the default timeout when testing on riscv rather than modifying the test framework?

Reply:
Currently the idea of the code is looping out there by setting up the fixed time (which is always > 0) to get the test involved to complete.  I am not sure what the default timeout is gonna be as it is hard to anticipate how long it takes to finish.

Reply:
The default could be set arbitrarily large, such as 24 hours or anything that makes sense.

Reply:
Will try to change the code  at /test/functional/cmdline_options_tester/src/TestConfigParser.java
```
				String timeout = (String)attributes.get("timeout");
			 _currentTest = new Test( id, timeout, _debugCmdOnTimeout ); <------- change it for riscv
```
to see how it goes this way.